config/routes.rb

```diff
@@ -9,6 +9,11 @@
   resources :sessions, only: %i[new create destroy]
   resource :password_reset, only: %i[new create edit update]
   resource :password, only: %i[edit update]
+  resources :email_confirmations, only: [] do
+    collection do
+      get :confirm, as: :confirm_email
+    end
+  end
```

Comment:
Totally semantic and optional, but for consistency with the other routes I think this is a great use case for something like:

```rb
resource :email_confirmation, only: :new
```

Would still use `email_confirmations_controller` and the new route would be `new_email_confirmation_url`